opm/input/eclipse/Schedule/Group/Group.cpp

```diff
@@ -30,15 +30,47 @@
 #include <fmt/format.h>
 
 #include <cmath>
+#include <cstdint>
 #include <stdexcept>
 #include <utility>
+#include <type_traits>
 
 namespace {
     bool is_defined(const double x)
     {
         return  x < (Opm::RestartIO::RstGroup::UNDEFINED_VALUE / 2);
     }
 
+    enum class GroupSatelliteStatus : std::uint_least8_t {
+        /// Group is not a satellite
+        None = UINT8_C(0),
+
+        /// Group has satellite production
+        Production = (UINT8_C(1) << UINT8_C(0)),
+    };
+
+    auto groupSatelliteStatusAsInteger(const GroupSatelliteStatus status)
+    {
+        return static_cast<std::underlying_type_t<GroupSatelliteStatus>>(status);
+    }
+
+    std::uint_least8_t setGroupSatelliteStatus(const std::uint_least8_t   current,
```

Comment:
append maybe? set can be a bit confusing since it sounds 'active'. then again this is an internal of course.